spec/javascripts/miq_application_spec.js

```diff
@@ -440,7 +440,7 @@ describe('miq_application.js', function() {
     context('failed login', function() {
       beforeEach(function() {
         // simulate failed authentication api call
-        spyOn(vanillaJsAPI, 'login').and.callFake(function() {
+        spyOn(window, 'miqAjaxAuth').and.callFake(function() {
```

Comment:
@himdel I have no clue what else could I do here to make this green as I don't understand the magic :cry:

Reply:
Aaah, sorry about that, `miqAjaxAuth` fails in a different way - it does `render :update` and calls *some* of those things that way.

But that probably means all that failure code in `miqAjaxAuth` is broken too now...

Looking..